### PR TITLE
cmake: do not build tests for Release build and cleanups

### LINKED ISSUE #2445: compile error with db/db_test_util.cc
Here is the build step

```
cd rocksdb
git pull origin master
mkdir build & cd build
cmake ..
make -j 12
```

then make throws errors

```
/home/tumh/rocksdb/db/db_test_util.cc: In destructor ‘virtual rocksdb::DBTestBase::~DBTestBase()’:
/home/tumh/rocksdb/db/db_test_util.cc:67:12: error: ‘rocksdb::SyncPoint’ has not been declared
   rocksdb::SyncPoint::GetInstance()->DisableProcessing();
            ^
/home/tumh/rocksdb/db/db_test_util.cc:68:12: error: ‘rocksdb::SyncPoint’ has not been declared
   rocksdb::SyncPoint::GetInstance()->LoadDependency({});
            ^
/home/tumh/rocksdb/db/db_test_util.cc:69:12: error: ‘rocksdb::SyncPoint’ has not been declared
   rocksdb::SyncPoint::GetInstance()->ClearAllCallBacks();
            ^
/home/tumh/rocksdb/db/db_test_util.cc: In member function ‘rocksdb::Options rocksdb::DBTestBase::GetOptions(int, const rocksdb::Options&, const rocksdb::anon::OptionsOverride&) const’:
/home/tumh/rocksdb/db/db_test_util.cc:278:12: error: ‘rocksdb::SyncPoint’ has not been declared
   rocksdb::SyncPoint::GetInstance()->ClearCallBack(
            ^
/home/tumh/rocksdb/db/db_test_util.cc:280:12: error: ‘rocksdb::SyncPoint’ has not been declared
   rocksdb::SyncPoint::GetInstance()->ClearCallBack(
            ^
/home/tumh/rocksdb/db/db_test_util.cc:454:16: error: ‘rocksdb::SyncPoint’ has not been declared
       rocksdb::SyncPoint::GetInstance()->SetCallBack(
                ^
/home/tumh/rocksdb/db/db_test_util.cc:459:16: error: ‘rocksdb::SyncPoint’ has not been declared
       rocksdb::SyncPoint::GetInstance()->SetCallBack(
                ^
/home/tumh/rocksdb/db/db_test_util.cc:464:16: error: ‘rocksdb::SyncPoint’ has not been declared
       rocksdb::SyncPoint::GetInstance()->EnableProcessing();
                ^
/home/tumh/rocksdb/db/db_test_util.cc: In member function ‘void rocksdb::DBTestBase::MoveFilesToLevel(int, int)’:
/home/tumh/rocksdb/db/db_test_util.cc:957:17: error: ‘class rocksdb::DBImpl’ has no member named ‘TEST_CompactRange’
       dbfull()->TEST_CompactRange(l, nullptr, nullptr, handles_[cf]);
                 ^
/home/tumh/rocksdb/db/db_test_util.cc:959:17: error: ‘class rocksdb::DBImpl’ has no member named ‘TEST_CompactRange’
       dbfull()->TEST_CompactRange(l, nullptr, nullptr);
                 ^
/home/tumh/rocksdb/db/db_test_util.cc: In member function ‘void rocksdb::DBTestBase::DumpFileCounts(const char*)’:
/home/tumh/rocksdb/db/db_test_util.cc:967:21: error: ‘class rocksdb::DBImpl’ has no member named ‘TEST_MaxNextLevelOverlappingBytes’
           dbfull()->TEST_MaxNextLevelOverlappingBytes());
                     ^
/home/tumh/rocksdb/db/db_test_util.cc: In member function ‘void rocksdb::DBTestBase::GenerateNewFile(int, rocksdb::Random*, int*, bool)’:
/home/tumh/rocksdb/db/db_test_util.cc:1008:15: error: ‘class rocksdb::DBImpl’ has no member named ‘TEST_WaitForFlushMemTable’
     dbfull()->TEST_WaitForFlushMemTable();
               ^
/home/tumh/rocksdb/db/db_test_util.cc:1009:15: error: ‘class rocksdb::DBImpl’ has no member named ‘TEST_WaitForCompact’
     dbfull()->TEST_WaitForCompact();
               ^
/home/tumh/rocksdb/db/db_test_util.cc: In member function ‘void rocksdb::DBTestBase::GenerateNewFile(rocksdb::Random*, int*, bool)’:
/home/tumh/rocksdb/db/db_test_util.cc:1020:15: error: ‘class rocksdb::DBImpl’ has no member named ‘TEST_WaitForFlushMemTable’
     dbfull()->TEST_WaitForFlushMemTable();
               ^
/home/tumh/rocksdb/db/db_test_util.cc:1021:15: error: ‘class rocksdb::DBImpl’ has no member named ‘TEST_WaitForCompact’
     dbfull()->TEST_WaitForCompact();
               ^
/home/tumh/rocksdb/db/db_test_util.cc: In member function ‘void rocksdb::DBTestBase::GenerateNewRandomFile(rocksdb::Random*, bool)’:
/home/tumh/rocksdb/db/db_test_util.cc:1033:15: error: ‘class rocksdb::DBImpl’ has no member named ‘TEST_WaitForFlushMemTable’
     dbfull()->TEST_WaitForFlushMemTable();
               ^
/home/tumh/rocksdb/db/db_test_util.cc:1034:15: error: ‘class rocksdb::DBImpl’ has no member named ‘TEST_WaitForCompact’
     dbfull()->TEST_WaitForCompact();

```

Any idea?

### PR DESCRIPTION
**Pull request by tchaikov**
fixes #2445

### CODE REVIEW

**Comment by riversand963** [comment]
Appveyor seems to be broken by the PR.

**Comment by tchaikov** [comment]
@riversand963 could you help review this change?

**Comment by tchaikov** [comment]
ping?

**Comment by tchaikov** [comment]
@riversand963 @siying , is there anything i can do to progress this change? i just found that #6098 was merged. it address this issue in a different way. but i think this this change is still relevant.

**Comment by riversand963** [comment]
Will take a look.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1062,69 +1069,69 @@ if(WITH_TESTS AND CMAKE_BUILD_TYPE STREQUAL "Debug")
...
-  foreach(sourcefile ${BENCHMARKS})
-    get_filename_component(exename ${sourcefile} NAME_WE)
-    add_executable(${exename}${ARTIFACT_SUFFIX} ${sourcefile}
-      $<TARGET_OBJECTS:testharness>)
-    target_link_libraries(${exename}${ARTIFACT_SUFFIX} gtest ${LIBS})
-  endforeach(sourcefile ${BENCHMARKS})
-
-  add_executable(db_bench${ARTIFACT_SUFFIX}
-    tools/db_bench.cc
-    tools/db_bench_tool.cc
```
Comment (siying) [inline CMakeLists.txt]: I'm a little bit confused why it can still build after removing this file.

Comment (tchaikov) [reply CMakeLists.txt]: my bad, added `db_bench_tool.cc` back . the reason it built is that gflags is not found by cmake as neither do we have `Findgflags.cmake` nor `{gflagsConfig,gflags-config}.cmake` in the (ancient) xenial CI node created by travis.

i added `Findgflags.cmake` to address this issue in this PR.

**Comment by siying** [comment]
Travis fails. Can you take a look?

**Comment by tchaikov** [comment]
@siying yes. fixed, rebased and repushed.

**Comment by siying** [comment]
Thank you for your contribution and your patient!

**Pull request closed without merging**